test/Microsoft.Framework.Runtime.Tests/ProjectFilesCollectionFacts.cs

```diff
@@ -59,6 +59,30 @@ public void FilePatternsWorkForArraysAreSet()
             Assert.Equal(new[] { "a.cs", @"foo.js" }, target.ResourcesPatterns);
         }
 
+
+        [Fact]
+        public void FilePatternsDoNotIncludeComments()
+        {
+            var rawProject = Deserialize(@"
+{
+    ""code"": [""*.cs"", /* FOO */ ""../*.cs""],
+    ""exclude"": [""buggy/*.*"", /* FOO */],
```

Comment:
Yeah the approach looks good to me. I saw @davidfowl looking at it and he didn't seem to freak out so I think this is fine.